src/main/scala/viper/silver/parser/ParseAst.scala

```diff
@@ -569,9 +580,10 @@ object POpApp {
   def pRes = PTypeVar(pResS)
 }
 
-case class PCall(func: PIdnUse, args: Seq[PExp], typeAnnotated: Option[PType] = None)(val pos: (Position, Position)) extends POpApp with PLocationAccess {
+case class PCall(func: PIdnUse, args: Seq[PExp], typeAnnotated: Option[PType] = None)(val pos: (Position, Position)) extends POpApp with PLocationAccess with PAssignTarget with PMacro {
```

Comment:
This subtyping relationship looks dubious to me. A call is a macro (call)? Shouldn't macro (call) be a type of call instead?

Reply:
The `PMacro` trait means that the thing might be a reference to a macro, same for `PIdnUse`. If we were to have two separate `PFnCall` and `PMacroCall` exprs (with a common `PCall` trait) then we wouldn't know which one to return at parsing.